From a62467b882d54d2cffcfe5484cedd860f171aacd Mon Sep 17 00:00:00 2001
From: Elon Zhang <zhangzj@rock-chips.com>
Date: Mon, 4 Nov 2019 17:45:10 +0800
Subject: [PATCH] ARM: dts: rk312x: move rng node from rk312x.dtsi to
 rk3128.dtsi

There is crypto hardware in RK3128 but NOT in RK3126C.

Change-Id: Ie5ab4e2565a34b7ea963f21793b46cafdf8a2c7f
Signed-off-by: Elon Zhang <zhangzj@rock-chips.com>
---
 arch/arm/boot/dts/rk3128.dtsi | 12 ++++++++++++
 arch/arm/boot/dts/rk312x.dtsi | 12 ------------
 2 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/arch/arm/boot/dts/rk3128.dtsi b/arch/arm/boot/dts/rk3128.dtsi
index a116a71c39a4..3dd1f2d6972d 100644
--- a/arch/arm/boot/dts/rk3128.dtsi
+++ b/arch/arm/boot/dts/rk3128.dtsi
@@ -43,6 +43,18 @@
 / {
 	compatible = "rockchip,rk3128";
 
+	rng: rng@100fc000 {
+		compatible = "rockchip,cryptov1-rng";
+		reg = <0x100fc000 0x4000>;
+		clocks = <&cru SCLK_CRYPTO>, <&cru HCLK_CRYPTO>;
+		clock-names = "clk_crypto", "hclk_crypto";
+		assigned-clocks = <&cru SCLK_CRYPTO>, <&cru HCLK_CRYPTO>;
+		assigned-clock-rates = <150000000>, <100000000>;
+		resets = <&cru SRST_CRYPTO>;
+		reset-names = "reset";
+		status = "disabled";
+	};
+
 	qos_ebc: qos@1012f080 {
 		compatible = "syscon";
 		reg = <0x1012f080 0x20>;
diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index 469ca5783adc..abe4486751ae 100644
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -465,18 +465,6 @@
 		};
 	};
 
-	rng: rng@100fc000 {
-		compatible = "rockchip,cryptov1-rng";
-		reg = <0x100fc000 0x4000>;
-		clocks = <&cru SCLK_CRYPTO>, <&cru HCLK_CRYPTO>;
-		clock-names = "clk_crypto", "hclk_crypto";
-		assigned-clocks = <&cru SCLK_CRYPTO>, <&cru HCLK_CRYPTO>;
-		assigned-clock-rates = <150000000>, <100000000>;
-		resets = <&cru SRST_CRYPTO>;
-		reset-names = "reset";
-		status = "disabled";
-	};
-
 	hevc: hevc_service@10104000 {
 		compatible = "rockchip,sub";
 		reg = <0x10104000 0x400>;
-- 
2.35.3

